evalml/model_understanding/graphs.py

```diff
@@ -471,7 +473,17 @@ def partial_dependence(pipeline, X, features, grid_resolution=100):
         ValueError: if the provided pipeline isn't fitted.
         ValueError: if the provided pipeline is a Baseline pipeline.
     """
+
     X = _convert_to_woodwork_structure(X)
+    # Dynamically set the grid resolution to the maximum number of categories
+    # in the categorical variables if there are more categories than resolution cells
+    try:
+        max_num_cats = max(X.select("categorical").describe().loc["nunique"])
+        if max_num_cats > grid_resolution:
```

Comment:
Curious, what happens when max_num_cats == grid_resolution?

Reply:
That's a good call, I should add in explicit test cases for this.  Let me fiddle with the test I have and add tests for max_num_cats < grid_resolution & max_num_cats == grid_resolution.

Reply:
So just to resolve this - what actually happens is the same issue as initially reported.  Only the case where grid resolution > max cats lets us get through and generate partial_dependence.